File: api/src/api/modules/organization/schemas/protocol.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class Activity(BaseModel):
    order: int
    name: str
    path: str
    has_time_limit: bool
    time_limit: int
    start_message: str
    end_message: str
    close_activity: bool
    show_timer: bool


class ProtocolRes(BaseModel):
    name: str
    activities: list[Activity]
    locked: bool
    created_at: datetime
    updated_at: datetime


class ActivityPostReq(BaseModel):
    name: str
    path: Optional[str] = ""
    has_time_limit: bool
    time_limit: int = 0
    start_message: Optional[str] = ""
    end_message: Optional[str] = ""
    close_activity: bool
    show_timer: bool


class ProtocolPostReq(BaseModel):
    name: str
    activities: list[ActivityPostReq]

File: ui/src/renderer/src/modules/organization/types/Protocol.ts
export type Activity = {
  order: number
  name: string
  path: string
  has_time_limit: boolean
  time_limit: number
  start_message: string
  end_message: string
  close_activity: boolean
  show_timer: boolean
}

export type Protocol = {
  name: string
  activities: Activity[]
  locked: boolean
  created_at: string
  updated_at: string
}

export type ActivityCreate = {
  name: string
  path: string
  has_time_limit: boolean
  time_limit: number
  start_message: string
  end_message: string
  close_activity: boolean
  show_timer: boolean
}

export type ProtocolCreate = {
  name: string
  activities: ActivityCreate[]
}
